src/core/src/op/convert.cpp

```diff
@@ -18,10 +19,18 @@ namespace convert {
 #define CONVERT_ET_LIST \
     boolean, bf16, f16, f32, f64, i4, i8, i16, i32, i64, u1, u2, u3, u4, u6, u8, u16, u32, u64, nf4, f8e4m3, f8e5m2
 
+#define CONVERT_FROM_NF4_ET_LIST bf16, f16, f32, f64, nf4, f8e4m3, f8e5m2
+
+#define CONVERT_FROM_NOT_FP_ET_LIST \
+    boolean, bf16, f16, f32, f64, i4, i8, i16, i32, i64, u1, u2, u3, u4, u6, u8, u16, u32, u64, f8e4m3, f8e5m2
+
 struct Evaluate : public element::NoAction<bool> {
     using element::NoAction<bool>::visit;
 
-    template <element::Type_t ET_IN, class TI = fundamental_type_for<ET_IN>>
+    // convert from FP (except NF4) to any other.
+    template <element::Type_t ET_IN,
```

Comment:
I've put a comment in the relevant ticket. In a nutshell, it would be better to support FP16->NF4.

Reply:
Changes applied to support FP16 -> NF4. Required to remove some test as they use not supported conversion by convert.

Reply:
But can we update the test instead of removing it to make sure that it works on the Python API level which is important for external users such as NNCF?

Reply:
Restored tests for python API and CPU. It looks like conversion from NF4 to F16 is required to make possible of decompression of constants (other types by adding additional convert).

But as CPU plugin on not ARM device always converts F16 to F32, I think would be better to add support in Convert to decompress NF4 to F32 directly without adding additional conversion.

Reply:
This is just a temporary situation. CPU will use BF16, INT8, of FP32 to unpack this type and it should be done on a micro-kernel level (not reference implementation). The reference implementation can have any type. My only concern is to provide an efficient way for compression from FP16/FP32 to NF4. Ideally, we should support both.

Reply:
@KodiaqQ mention in ticket that these changes makes no regression. 
I think PR can be merged, @AlexKoff88 can you confirm?